src/components/application_manager/test/commands/mobile/show_constant_tbt_test.cc

```diff
@@ -0,0 +1,402 @@
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <string>
+
+#include "application_manager/commands/mobile/show_constant_tbt_request.h"
+#include "application_manager/commands/command_request_test.h"
+#include "application_manager/policies/mock_policy_handler_interface.h"
+#include "application_manager/message_helper.h"
+#include "gtest/gtest.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace mobile_commands_test {
+namespace show_constant_tbt_request {
+
+namespace am = application_manager;
+using am::commands::ShowConstantTBTRequest;
+using ::utils::SharedPtr;
+using ::testing::_;
+using ::testing::Return;
+using ::testing::ReturnRef;
+
+using policy_test::MockPolicyHandlerInterface;
+
+namespace {
+const uint32_t kAppId = 1u;
+const uint32_t kConnectionKey = 2u;
+const std::string kNavigationText1 = "Some text1";
+const std::string kNavigationText2 = "Some text2";
+const std::string kEta = "Some time";
+const std::string kTotalDistance = "Some distance";
+const std::string kTimeToDestination = "Some time";
+const std::string kTutnIconValue = "0x1A3F";
+const std::string kNextTutnIconValue = "0x1A40";
+}  // namespace
+
+class ShowConstantTBTRequestTest
+    : public CommandRequestTest<CommandsTestMocks::kIsNice> {
+ public:
+  MessageSharedPtr CreateMessageWithParams() {
+    MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+    (*msg)[am::strings::params][am::strings::connection_key] = kConnectionKey;
+
+    smart_objects::SmartObject msg_params =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+    msg_params[am::strings::turn_icon][am::strings::value] = kTutnIconValue;
+    msg_params[am::strings::next_turn_icon][am::strings::value] =
+        kNextTutnIconValue;
+    msg_params[am::strings::navigation_text_1] = kNavigationText1;
+    msg_params[am::strings::navigation_text_2] = kNavigationText2;
+    msg_params[am::strings::eta] = kEta;
+    msg_params[am::strings::total_distance] = kTotalDistance;
+    msg_params[am::strings::time_to_destination] = kTimeToDestination;
+
+    smart_objects::SmartObject soft_button =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+    soft_button[am::strings::type] = am::mobile_api::SoftButtonType::SBT_TEXT;
+    soft_button[am::strings::text] = "some text";
+    soft_button[am::strings::soft_button_id] = 15u;
+
+    smart_objects::SmartObject soft_buttons =
+        smart_objects::SmartObject(smart_objects::SmartType_Array);
+    soft_buttons[0] = soft_button;
+
+    msg_params[am::strings::soft_buttons] = soft_buttons;
+    (*msg)[am::strings::msg_params] = msg_params;
+
+    return msg;
+  }
+
+  MessageSharedPtr CreateExpectedMessage() {
+    MessageSharedPtr msg = CreateMessage(smart_objects::SmartType_Map);
+
+    smart_objects::SmartObject msg_params =
+        smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+    int32_t index = 0;
+
+    msg_params[am::hmi_request::navi_texts][index]
```

Comment:
@BSolonenko please extract msg_params[am::hmi_request::navi_texts] to reference variable